watcher/local/watcher.go

```diff
@@ -33,6 +34,16 @@ const (
 )
 
 type (
+	// closer is the interface that wraps the close method.
+	closer interface {
+		close() error
+	}
+
+	// Closer is the interface that wraps the Close method.
+	Closer interface {
```

Comment:
Im not really happy with the ad-hoc redefenition of a public `Closer` interface here.  
Please consider using `[pkg/sync] Closer` which also gives so some guarantees as well.

Reply:
Just noticed the interface itself need not be public. It is just used to store the pub-sub interfaces in the channel instance (of watcher) to access them while closing.

I have renamed this interface. See PR #216.